scvi/inference/posterior.py

```diff
@@ -208,15 +208,14 @@ def sample_scale_from_batch(self, n_samples, batchid=None, selection=None):
         else:
             if selection.dtype is np.dtype('bool'):
                 selection = np.asarray(np.where(selection)[0].ravel())
+        old_loader = self.data_loader
```

Comment:
Why are we creating a new variable called `old_loader`? Is there a way to make this whole block of code simpler? I'm not sure a few lines below why we'd overwrite `self.data_loader` once for each batch id in the first place.

Reply:
(I may just be missing something, but if the whole reason for introducing `old_loader` is because `self.data_loader` is getting overwritten below unnecessarily, it'd be better to avoid all the unnecessary assignment.)

Reply:
The purpose of the function is to impute the scales for each batch individually. To do this, you must sample the data in only one batch at a time. The `DataLoader' class is exactly designed for this task as soon as you change its `Sampler' to sample only from a subset of inputs. 

There was a huge problem with the way it was done before. It was overwriting `self.data_loader' without restoring it as it was before. Indeed, he restored it with a RandomSampler, but in many cases - as after calling .sequential on the Posterior - the sampler was not Random but Sequential. When I was asking for the first input, I was receiving a random one by the Sampler.

Using `old_loader` fixes this and restores the DataLoader: no more side effect!